java/vector/src/main/codegen/templates/UnionWriter.java

```diff
@@ -110,6 +143,26 @@ public ListWriter asList() {
     return getListWriter();
   }
 
+  private MapWriter getMapWriter() {
+    // returns existing writer
+    Preconditions.checkNotNull(mapWriter);
```

Comment:
Should this also create a writer if there is no existing?

Reply:
I think this is another case of, we need keysSorted in order to create the writer so we can't construct a new writer, but must rely on an existing one.

Reply:
I've updated the PR to assume `keysSorted = false` by default.